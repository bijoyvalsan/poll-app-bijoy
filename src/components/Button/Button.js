import styled from "styled-components";

export const Button = styled.button`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: auto;
  max-height: 50px;
  margin: 0px;
  cursor: pointer;
  font-size: 16px;
  line-height: 24px;
  text-align: center;
  text-decoration: none;
  font-weight: 500;
  border-width: 1px;
  border-style: solid;
  background-color: black;
  color: white;
  padding: 5px;
  &:hover {
    opacity: 0.6;
  }
  &:disabled {
    opacity: 0.3;
    cursor: inherit;
  }
`;
