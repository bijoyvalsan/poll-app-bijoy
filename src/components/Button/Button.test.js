import React from "react";
import { render } from "@testing-library/react";

import { Button } from "./Button";

describe("Button Component", () => {
  it("should render with default button styles", () => {
    const { asFragment } = render(<Button>Test Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
