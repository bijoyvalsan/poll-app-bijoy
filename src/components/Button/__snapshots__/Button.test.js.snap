// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render with default button styles 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnxRM kWuiVc"
  >
    Test Button
  </button>
</DocumentFragment>
`;
